commit b160d0428245fe4edb3e6e7d61ec01dfcc20c64c (refs/changes/25/4625/4)
Author: Trent Mick <trentm@gmail.com>
Date:   2018-08-02T14:09:24-07:00 (1 year, 2 months ago)
    
    TRITON-621 mockcloud delegate dataset support (dummy backend will use 'mockcloudRoot' mdata for mock server data, if available)
    Reviewed by: Josh Wilsdon <josh@wilsdon.ca>
    Approved by: Josh Wilsdon <josh@wilsdon.ca>

diff --git a/lib/backends/dummy/common.js b/lib/backends/dummy/common.js
index 5c8efaa..e64c67c 100644
--- a/lib/backends/dummy/common.js
+++ b/lib/backends/dummy/common.js
@@ -21,7 +21,19 @@ var fs = require('fs');
 var assert = require('assert-plus');
 var async = require('async');
 
-var SERVER_ROOT = '/opt/custom/virtual/servers';
+var mockcloudRoot;
+try {
+    mockcloudRoot = child_process
+        .execSync('/usr/sbin/mdata-get mockcloudRoot', {encoding: 'utf8'})
+        .trim();
+} catch (err) {
+    // The old default for backward compatibility.
+    mockcloudRoot = '/opt/custom/virtual';
+    console.warn('warning: cn-agent dummy backend could not get ' +
+        '"mockcloudRoot" dir from mdata, using default %s: %s',
+        mockcloudRoot, err);
+}
+var SERVER_ROOT = mockcloudRoot + '/servers';
 
 function mdataGet(key, callback) {
     assert.string(key, 'key');
diff --git a/lib/backends/dummy/tools/create-server.js b/lib/backends/dummy/tools/create-server.js
index e680291..e8e8ec1 100644
--- a/lib/backends/dummy/tools/create-server.js
+++ b/lib/backends/dummy/tools/create-server.js
@@ -286,7 +286,7 @@ function writeSysinfo(ctx, callback) {
 
     fs.writeFile(filename, data, function _wroteSysinfo(err) {
         if (!err) {
-            console.log('wrote sysinfo');
+            console.log('wrote sysinfo (%s)', filename);
         }
 
         callback(err);
diff --git a/package.json b/package.json
index 541c39f..3897009 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "cn-agent",
   "description": "Triton Compute Node Agent",
-  "version": "2.3.0",
+  "version": "2.3.1",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
